src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,350 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import FastImage from 'react-native-fast-image'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import SkeletonPlaceholder from 'src/components/SkeletonPlaceholder'
+import Touchable from 'src/components/Touchable'
+import InfoIcon from 'src/icons/InfoIcon'
+import OpenLinkIcon from 'src/icons/OpenLinkIcon'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import { Nft } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import Logger from 'src/utils/Logger'
+import networkConfig from 'src/web3/networkConfig'
+
+const TAG = 'nfts/NftsInfoCarousel'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.NftsInfoCarousel>
+
+function scaleImageHeight(originalWidth: number, originalHeight: number, targetWidth: number) {
+  const aspectRatio = originalWidth / originalHeight
+  return targetWidth / aspectRatio
+}
+
+const NftsLoadErrorScreen = () => {
+  const { t } = useTranslation()
+  return (
+    <SafeAreaView
+      testID="NftsInfoCarousel/NftsLoadErrorScreen"
+      style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}
+    >
+      <Text>{t('nftInfoCarousel.noNftsFound')}</Text>
+    </SafeAreaView>
+  )
+}
+
+const ThumbnailImagePlaceholder = () => {
+  return (
+    <SkeletonPlaceholder backgroundColor={colors.gray2} highlightColor={colors.white}>
+      <View style={styles.skeletonImageThumbnailPlaceHolder} />
+    </SkeletonPlaceholder>
+  )
+}
+
+const NftThumbnail = ({
+  nft,
+  isActive,
+  onPress,
+  error,
+}: {
+  nft: Nft
+  isActive: boolean
+  onPress: () => void
+  error?: boolean
+}) => {
+  const [loading, setLoading] = useState(true)
+  return (
+    <Touchable
+      style={[
+        isActive ? styles.nftThumbnailSelected : styles.nftThumbnailUnSelected,
+        error && styles.errorThumbnail,
+      ]}
+      borderless={false}
+      onPress={onPress}
+      testID={`NftsInfoCarousel/NftThumbnail/${nft.contractAddress}-${nft.tokenId}`}
+    >
+      {error ? (
```

Comment:
I was confused why media was an array first at too, but it makes more sense given to context here: https://github.com/valora-inc/nft-indexer/pull/331. Right now it's just an array with one object but it could also contain entries for animation_url, video_url, mp4_url etc.